### PR TITLE
Mlibz 1612 type error push unregister

### PR DESCRIPTION
**Pull request by thomasconner**
### Description
This PR fixes a `TypeError` that is encountered when unregistering a device for push notifications. This PR also adds unit tests for the `Push` class.

### Changes
- Check if the stored pushDeviceConfig is defined before accessing properties.
- Add unit tests for the `Push` class.

### CODE REVIEW

File: src/push.js [hunk 1]
```diff
@@ -132,11 +140,17 @@ class Push extends EventEmitter {
       })
       .then(() => {
         const user = User.getActiveUser(this.client);
-        const key = user ? `${this.pathname}_${user._id}` : `${this.pathname}__${options.userId}`;
+        const key = user ? `${this.pathname}_${user._id}` : `${this.pathname}_${options.userId}`;
```
Comment (tejasranade) [inline src/push.js]: @thomasconner just checking if this is a deliberate change.

File: src/push.js [hunk 2]
```diff
@@ -154,12 +168,13 @@ class Push extends EventEmitter {
...
             deviceId: deviceId,
-            userId: user ? null : options.userId
+            userId: user ? undefined : options.userId
           },
           timeout: options.timeout,
           client: this.client
         });
-        return request.execute();
+        return request.execute()
+          .then(response => response.data);
```
Comment (tejasranade) [inline src/push.js]: @thomasconner Does this change the API signature for the app developer?

File: src/push.js (hunk 1 above)
Comment (thomasconner) [inline src/push.js]: Yes it was. The double `__` was a typo.

File: src/push.js (hunk 2 above)
Comment (thomasconner) [inline src/push.js]: No this does not.

**Pull request merged**